src/OrchardCore.Modules/OrchardCore.Markdown/GraphQL/MarkdownBodyQueryObjectType.cs

```diff
@@ -12,19 +17,25 @@ public MarkdownBodyQueryObjectType(IStringLocalizer<MarkdownBodyQueryObjectType>
             Description = T["Content stored as Markdown. You can also query the HTML interpreted version of Markdown."];
 
             Field("markdown", x => x.Markdown, nullable: true)
-                .Description("the markdown value")
-                .Type(new StringGraphType())
-                ;
+                .Description(T["the markdown value"])
+                .Type(new StringGraphType());
 
-            Field("html", x => ToHtml(x.Markdown), nullable: true)
-                .Description("the HTML representation of the markdown content")
-                .Type(new StringGraphType())
-                ;
+            Field<StringGraphType>()
+                .Name("html")
+                .Description(T["the HTML representation of the markdown content"])
+                .ResolveAsync(ToHtml);
         }
 
-        private static string ToHtml(string markdown)
+        private static async Task<object> ToHtml(ResolveFieldContext<MarkdownBodyPart> ctx)
         {
-            return Markdig.Markdown.ToHtml(markdown ?? "");
+            var context = (GraphQLContext) ctx.UserContext;
+            var liquidTemplateManager = context.ServiceProvider.GetService<ILiquidTemplateManager>();
+
+            var markdown = ctx.Source.Markdown;
+            var templateContext = new TemplateContext();
```

Comment:
I think we need to call a special service (Contextualize)

Reply:
Happy to do that for HtmlBodyQueryObjectType.cs since there is some duplication and complexity in building the TemplateContext. The others I could remove the duplication by creating an extension such as the below?
```
        ...
            Field<StringGraphType>()
                .Name("html")
                .Description(T["the HTML representation of the markdown content"])
                .ResolveAsync(ToHtml);
        }

        private static async Task<object> ToHtml(ResolveFieldContext<MarkdownBodyPart> ctx)
        {
            return await ctx.ToHtmlAsync(ctx.Source.Markdown);
        }
    }

    public static class ResolveFieldContextExtensions
    {
        public static async Task<string> ToHtmlAsync<TSource>(this ResolveFieldContext<TSource> ctx, string markdown)
        {
            var context = (GraphQLContext) ctx.UserContext;
            var liquidTemplateManager = context.ServiceProvider.GetService<ILiquidTemplateManager>();

            var templateContext = new TemplateContext();
            markdown = await liquidTemplateManager.RenderAsync(markdown, templateContext);

            return Markdig.Markdown.ToHtml(markdown);
        }
    }
```

Reply:
Just for infos, if @sebastienros is talking about `TemplateContext.ContextualizeAsync()`, now this is automatically done through `liquidTemplateManager.RenderAsync()` since a PR adding a more complete contextualization in any context.

Reply:
Thanks for the additional info @jtkech, I was guessing at what sebastien meant and got it wrong, so looks like we don't need to worry about `ContextualizeAsync`. I'm still happy to reduce the code duplication if necessary.

Reply:
No, it's all I wanted, thanks